WickedEngine/wiBacklog.cpp

```diff
@@ -45,62 +50,259 @@ namespace wi::backlog
 	std::mutex historyLock;
 	std::string logfile_path = "";
 
+	struct AsyncWriter
+	{
+		std::thread writerThread;
+		std::mutex queueMutex;
+		std::condition_variable queueCondition;
+		std::deque<std::string> writeQueue;
+		std::atomic<bool> running{ false };
+		std::atomic<bool> initialized{ false };
+		std::atomic<uint32_t> autoFlushInterval{ 1000 };
+		std::chrono::steady_clock::time_point lastFlushTime;
+		std::ofstream logFileStream;
+		std::string currentLogFilePath;
+		std::mutex flushMutex; // Synchronous flush requests
+
+		void Start(const std::string& filepath)
+		{
+			if (initialized.exchange(true))
+			{
+				return;
+			}
+
+			currentLogFilePath = filepath;
+			running = true;
+			lastFlushTime = std::chrono::steady_clock::now();
+
+			logFileStream.open(filepath, std::ios::binary | std::ios::trunc);
+			if (!logFileStream.is_open())
+			{
+				wi::helper::DebugOut("Failed to open log file: " + filepath + "\n", wi::helper::DebugLevel::Error);
+			}
+
+			writerThread = std::thread([this]() {
+				WriterLoop();
+			});
+		}
+
+		void Stop()
+		{
+			if (!initialized.load())
+			{
+				return;
+			}
+
+			running = false;
+			queueCondition.notify_all();
+
+			if (writerThread.joinable())
+			{
+				writerThread.join();
+			}
+
+			// Final flush of any remaining entries
+			FlushQueue();
+
+			if (logFileStream.is_open())
+			{
+				logFileStream.flush();
+				logFileStream.close();
+			}
+
+			initialized = false;
+		}
+
+		void WriterLoop()
+		{
+			while (running.load())
+			{
+				std::unique_lock lock(queueMutex);
+
+				// Wait for new entries or timeout for periodic flush
+				auto timeout = std::chrono::milliseconds(100);
+				queueCondition.wait_for(lock, timeout, [this]() {
+					return !writeQueue.empty() || !running.load();
+				});
+
+				if (!running.load() && writeQueue.empty())
+					break;
+
+				// Process all queued entries
+				std::deque<std::string> localQueue;
+				std::swap(localQueue, writeQueue);
+				lock.unlock();
+
+				WriteEntries(localQueue);
+
+				// Check for auto-flush interval
+				auto now = std::chrono::steady_clock::now();
```

Comment:
we also have wi::Timer thing that could be used, I think that would be better with more understadeable functions.